frontend/src/metabase-lib/references.ts

```diff
@@ -106,6 +106,7 @@ export const getDimensionReferenceWithoutOptions = (
 export const BASE_DIMENSION_REFERENCE_OMIT_OPTIONS = [
   "temporal-unit",
   "binning",
+  "base-type",
 ];
```

Comment:
I'm worried this might blow up. Why do we need it?

Reply:
yeah, I was a little unsure about this, but this is needed to make MLv1 and MLv2 columns match up, without it, a filter created using a popover will have a column like: `['field', 99, { "base-type": boolean } ]` and the filterPopover will be looking for `['field', 99, null ]` and the `isSameBaseDimension()` check will be false, so they won't match up. 😢

Reply:
I think this should be safe when it's used for comparison, we're not actually mutating any queries that we send to be executed.